apps/cow-tools/src/app/milkman/(components)/SingleOrderForm/Twap.tsx

```diff
@@ -0,0 +1,95 @@
+import { zodResolver } from "@hookform/resolvers/zod";
+import { useEffect } from "react";
+import { Controller, FieldValues, useForm } from "react-hook-form";
+
+import { Checkbox } from "#/components/Checkbox";
+import { Input } from "#/components/Input";
+import { Select, SelectItem } from "#/components/Select";
+import { Form } from "#/components/ui/form";
+import { orderTwapSchema } from "#/lib/schema";
+
+import { TransactionStatus } from "../../utils/type";
+import { FormFooter } from "./Footer";
+
+export enum TWAP_DELAY_OPTIONS {
+  TEN_MINUTES = "10 minutes",
+  THIRTY_MINUTES = "30 minutes",
+  ONE_HOUR = "1 hour",
+  THREE_HOURS = "3 hours",
+  TWELVE_HOURS = "12 hours",
+  ONE_DAY = "1 day",
+}
+
+export const TwapDelayValues = {
+  [TWAP_DELAY_OPTIONS.TEN_MINUTES]: 10 * 60,
+  [TWAP_DELAY_OPTIONS.THIRTY_MINUTES]: 30 * 60,
+  [TWAP_DELAY_OPTIONS.ONE_HOUR]: 60 * 60,
+  [TWAP_DELAY_OPTIONS.THREE_HOURS]: 3 * 60 * 60,
+  [TWAP_DELAY_OPTIONS.TWELVE_HOURS]: 12 * 60 * 60,
+  [TWAP_DELAY_OPTIONS.ONE_DAY]: 24 * 60 * 60,
+};
+
+export function FormTwap({
+  onSubmit,
+  defaultValues,
+}: {
+  onSubmit: (data: FieldValues) => void;
+  defaultValues?: FieldValues;
+}) {
+  const form = useForm<typeof orderTwapSchema._type>({
+    resolver: zodResolver(orderTwapSchema),
+  });
+
+  const { register, setValue, watch, control } = form;
+
+  useEffect(() => {
+    register("isTwapNeeded");
+    register("delay");
+    register("numberOfOrders");
+    setValue("isTwapNeeded", defaultValues?.isTwapNeeded);
+    setValue("delay", defaultValues?.delay);
+    setValue("numberOfOrders", defaultValues?.numberOfOrders);
+  }, []);
```

Comment:
just curious about something: doesn't registering the fields in each input work, adding the defaults to `useForm`? That's the usual way I'd implement this, but I wonder if it might've not worked for what you needed here

Reply:
I think that the issue was that the input fields aren't rendered when isTwapNeeded is equal to false. We can change it by letting them just disabled, what do you think?